### PR TITLE
online payment fraud detection using ml algo 

### PR DESCRIPTION
**Pull request by sharath4444**
# Online Payment Fraud Detection using Machine Learning
=====================================================

This project aims to develop a machine learning model that can detect online payment fraud with high accuracy. The model is trained on a dataset containing historical information about fraudulent transactions.

## Table of Contents
-----------------

*   [Dataset](#dataset)
*   [Model Architecture](#model-architecture)
*   [Training and Evaluation](#training-and-evaluation)
*   [Features](#features)
*   [Contributing](#contributing)

## Dataset
---------

The dataset used for this project is obtained from Kaggle and contains the following features:

*   `step`: represents a unit of time where 1 step equals 1 hour
*   `type`: type of online transaction
*   `amount`: the amount of the transaction
*   `nameOrig`: customer starting the transaction
*   `oldbalanceOrg`: balance before the transaction
*   `newbalanceOrig`: balance after the transaction
*   `nameDest`: recipient of the transaction
*   `oldbalanceDest`: initial balance of recipient before the transaction
*   `newbalanceDest`: the new balance of recipient after the transaction
*   `isFraud`: fraud transaction (target variable)

## Model Architecture
-------------------

The machine learning model used for this project is a **Decision Tree Classifier**. The model is trained on the dataset to predict the `isFraud` target variable.

### Code

Here's an example code snippet for the model architecture:
```python
import pandas as pd
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import train_test_split

# Load the dataset
data = pd.read_csv('fraud_data.csv')

# Preprocess the data
X = data.drop(['isFraud'], axis=1)
y = data['isFraud']

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Train the model
model = DecisionTreeClassifier()
model.fit(X_train, y_train)

# Evaluate the model
accuracy = model.score(X_test, y_test)
print(f'Model accuracy: {accuracy:.3f}')

### CODE REVIEW

**Comment by sharath4444** [comment]
please add me a label of hacktoberfest
